implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -151,22 +156,40 @@ public void export(List<Meter> meters) {
             });
         }
 
+        // if the config to export metadata is turned off, the seenMetadata map will be
+        // null.
+        if (seenMetadata != null) {
+            seenMetadata.values().forEach(line -> {
+                if (line == null) {
+                    return;
+                }
+
+                batch.add(line);
+                if (batch.size() == partitionSize) {
+                    send(batch);
+                    batch.clear();
+                }
```

Comment:
Nit: this is the exact code as above, maybe refactor it into a method like `sendBatch`?

Reply:
I extracted it to a method named `addLineToBatchAndSendWhenBatchIsFull`. Not sure which is worse, having the same code here twice or having a method with that name 😅

Reply:
maybe `trySendBatch`? Honestly, `sendBatch` would be enough I think. If one is interested in what it does, looking at it will be pretty obvious. could also just leave it duplicated, it's not a big deal :)

Reply:
I changed it in ed2f7c881e531d43a950a66a0a1be9f557b8b98a. It's clear, I agree, but I also think having the method say what it actually does is important. I add the line to the batch outside the method now, and the method just sends the batch once its full.